File: backend/pokerogue/src/main/java/com/pokerogue/external/pokemon/dto/pokemon/PokemonDetail.java
package com.pokerogue.external.pokemon.dto.pokemon;

import com.pokerogue.external.pokemon.dto.DataUrl;

public record PokemonDetail(
        String name,
        double weight,
        double height,
        DataUrl species,
        int hp,
        int attack,
        int defense,
        int speed,
        int specialAttack,
        int specialDefense,
        int totalStats
) {
}

File: android/app/src/main/java/poke/rogue/helper/data/model/PokemonDetail.kt
package poke.rogue.helper.data.model

import poke.rogue.helper.remote.dto.response.pokemon.PokemonDetailResponse
import poke.rogue.helper.remote.dto.response.type.PokemonTypeResponse

data class PokemonDetail(
    val pokemon: Pokemon,
    val stats: List<Stat>,
    val abilities: List<String>,
    val height: Float,
    val weight: Float,
)

fun PokemonDetailResponse.toData(id: Long): PokemonDetail {
    return PokemonDetail(
        Pokemon(
            id = id,
            dexNumber = dexNumber,
            name = name,
            imageUrl = imageUrl,
            types = types.map(PokemonTypeResponse::toData),
        ),
        stats =
            listOf(
                Stat("total", totalStats),
                Stat("hp", hp),
                Stat("attack", attack),
                Stat("defense", defense),
                Stat("specialAttack", specialAttack),
                Stat("specialDefense", specialDefense),
                Stat("speed", speed),
            ),
        abilities = abilityNames,
        height = height,
        weight = weight,
    )
}
